src/test/java/net/solmis/cityparking/TicketTest.java

```diff
@@ -0,0 +1,38 @@
+package net.solmis.cityparking;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class TicketTest {
+
+    @Test
+    public void saveShouldChangeIdToPositive() {
+        Ticket ticket = Ticket.createTicketAndSetStartTimestamp(null);
+        assertEquals(0, ticket.getId());
+        ticket.save();
+        assertNotEquals(0, ticket.getId());
+    }
+
+    @Test
+    public void createdTicketShouldBeActive() {
+        Ticket ticket = Ticket.createTicketAndSetStartTimestamp(null);
+        assertTrue(ticket.isActive());
+    }
+
+    @Test
+    public void shouldBeInactiveAfterSettingTimestamp() {
+        Ticket ticket = Ticket.createTicketAndSetStartTimestamp(null);
+        ticket.setEndTimestamp();
+        assertFalse(ticket.isActive());
+    }
+
+    @Test
+    public void parkingTimeShouldBeZeroOrPositive() {
+        Ticket ticket = Ticket.createTicketAndSetStartTimestamp(null);
+        assertTrue(ticket.getParkingTimeInSeconds() >= 0);
+        ticket.setEndTimestamp();
+        assertTrue(ticket.getParkingTimeInSeconds() >= 0);
```

Comment:
czy to dobry sposób testowania upływu czasu?

Reply:
Ciężko to nazwać sposobem testowania upływu czasu. Raczej chodziło o sprawdzenie, czy zarówno w momencie po stworzeniu biletu, jak i jego zatrzymaniu pobieranie czasu postoju nie zwróci bezsensownych wartości (ujemnych). Np. policzenie czasu end_time - start_time dla nieprawdziwego end_time w przypadku, gdy parkowanie jeszcze nie zostało zastopowane. Być może powinienem inaczej nazwać ten test lub w ogóle go nie pisać.